src/synapse/workload.py

```diff
@@ -366,6 +366,42 @@ def enable_allow_public_rooms_over_federation(container: ops.Container) -> None:
         raise WorkloadError(str(exc)) from exc
 
 
+def _create_ip_range_whitelist(charm_state: CharmState) -> list[str]:
+    """Format IP range whitelist.
+
+    Args:
+        charm_state: Instance of CharmState.
+
+    Returns:
+        IP range whitelist as expected by Synapse or None.
+    """
+    # The field is validated in enable_ip_range_whitelist
+    whitelist = charm_state.synapse_config.ip_range_whitelist
+    return [item.strip() for item in whitelist.split(",")]  # type: ignore[union-attr]
```

Comment:
Since we're doing the logic of splitting and stripping whitespace, should there be a unit test covering:

1. empty list
2. single item list
3. multiple item list
4. item with leading whitespace
5. item with trailing whitespace